pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -115,7 +117,18 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	// Watch for changes to primary resource BareMetalHost
 	err = c.Watch(&source.Kind{Type: &metal3v1alpha1.BareMetalHost{}},
-		&handler.EnqueueRequestForObject{})
+		&handler.EnqueueRequestForObject{}, predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldHost := e.ObjectOld.(*metal3v1alpha1.BareMetalHost)
+				newHost := e.ObjectNew.(*metal3v1alpha1.BareMetalHost)
+
+				if oldHost.Status.ErrorCount != newHost.Status.ErrorCount {
```

Comment:
AFAICS no, but I cannot exclude it completely yet. The filter simply skips the current update when a change has been detected for ErrorCount (and btw it would be better to limit the condition to a `>` to avoid skipping a `ClearError` event).
My point is that an increment in the ErrorCount field indicates that an action error happened, so that we'd like to delay the next reconcile. Even though the current update is skipped (and this is only due the writing of the ErrorCount field) another reconcile is guaranteed to be scheduled in the next future (based on the backoff calculation)

Reply:
OK, I think the logic of this implementation is reasonable.

Reply:
I was worried about whether we should be ignoring the changes if there are other differences apart from the error fields, but I think I've satisfied myself that this should be sufficient. If we need requeues we should be requesting them and not relying on our own writes to the object to trigger the Watch.